File: api/jsquick/log.js
const { blue, yellow, red, blueBright, green } = require("console-log-colors");
const { name, version } = require("../package.json");

class log {
  logTitle() {
    console.log(blueBright(`${name}  ${version}`));
  }
  logWarning(str) {
    console.log(`${yellow("WARN")} ${str}`);
  }
  logError(str) {
    console.log(`${red("ERRO")} ${str}`);
  }
  logInfo(str) {
    console.log(`${blue("INFO")} ${str}`);
  }
  logSuccess(str) {
    console.log(`${green("SUCC")} ${str}`);
  }
}
module.exports = new log();

File: Log/Log.cpp
//
// Created by georgeckito on 7/19/2024.
//

#include "Log.h"

#include <iostream>

#include "../color.hpp"

namespace Log {
    void logItem(std::string name, std::string path) {
        std::cout << "  " << dye::light_yellow(name) << " -> " << path << std::endl;
    }
    void logTitle() {
        std::cout << dye::aqua("jsquick v1.0.0") << std::endl;
    }
    void logWarn(const std::string &text) {
        std::cout << dye::yellow("WARN") << " " << text << std::endl;
    }
    void logError(const std::string &text) {
        std::cout << dye::red("ERRO") << " " << text << std::endl;
    }
    void logInfo(const std::string &text) {
        std::cout << dye::blue("INFO") << " " << text << std::endl;
    }
    void logSUCC(const std::string &text) {
        std::cout << dye::green("SUCC") << " " << text << std::endl;
    }
}
